models/DataObject/ClassDefinition/Data/Multiselect.php

```diff
@@ -279,6 +279,46 @@ public function checkValidity(mixed $data, bool $omitMandatoryCheck = false, arr
         if (!is_array($data) && !empty($data)) {
             throw new Model\Element\ValidationException("Invalid multiselect data on field [ {$this->getName()} ]");
         }
+
+        if (is_array($data)) {
+            // Ensure options providers are resolved
+            if ($this->getOptions() === null) {
+                $this->enrichFieldDefinition($params['context'] ?? []);
+            }
+
+            foreach ($data as $value) {
+                if (!$this->isValidOption($value)) {
+                    throw new Model\Element\ValidationException(
+                        sprintf("Invalid multiselect option '%s' on field [ %s ]", $value, $this->getName())
+                    );
+                }
+            }
+        }
+    }
+
+    /**
+     * Validates if the provided data matches any of the available options.
+     *
+     */
+    public function isValidOption(mixed $data): bool
+    {
+        // If no options are defined yet, skip the validation
+        if (empty($this->getOptions())) {
```

Comment:
We should avoid using empty here. See https://localheinz.com/articles/2023/05/10/avoiding-empty-in-php/
```suggestion
        if (!$this->getOptions()) {
```